src/components/ha-file-upload.ts

```diff
@@ -44,73 +55,109 @@ export class HaFileUpload extends LitElement {
+                      .path=${this.icon || mdiFileUpload}
+                    ></ha-svg-icon>
+                    <span class="header"
+                      >${this.label ||
+                      this.hass?.localize("ui.components.file-upload.label", {
+                        browse: html`<span class="highlight"
+                          >${this.hass?.localize(
+                            "ui.components.file-upload.browse"
+                          )}</span
+                        >`,
+                      })}</span
+                    >
+                    <span class="secondary" id="label">${this.secondary}</span>`
+                : typeof this.value === "string"
+                ? html`<div class="row">
+                    <div class="value" @click=${this._openFilePicker}>
+                      <ha-svg-icon
+                        slot="icon"
+                        .path=${this.icon || mdiFileUpload}
+                      ></ha-svg-icon>
+                      ${this.value}
+                    </div>
                     <ha-icon-button
-                      @click=${this._openFilePicker}
-                      .path=${this.icon}
+                      slot="suffix"
+                      @click=${this._clearValue}
+                      .label=${this.hass?.localize("ui.common.delete") ||
+                      "Delete"}
+                      .path=${mdiDelete}
                     ></ha-icon-button>
-                  </span>`
-                : ""}
-              <div class="value">${this.value}</div>
+                  </div>`
+                : (this.value instanceof FileList
+                    ? Array.from(this.value)
+                    : ensureArray(this.value)
+                  ).map(
+                    (file) =>
+                      html`<div class="row">
+                        <div class="value" @click=${this._openFilePicker}>
+                          <ha-svg-icon
+                            slot="icon"
+                            .path=${this.icon || mdiFileUpload}
+                          ></ha-svg-icon>
+                          ${file.name} - ${bytesToString(file.size)}
+                        </div>
+                        <ha-icon-button
+                          slot="suffix"
+                          @click=${this._clearValue}
+                          .label=${this.hass?.localize("ui.common.delete") ||
+                          "Delete"}
+                          .path=${mdiDelete}
+                        ></ha-icon-button>
+                      </div>`
+                  )}
               <input
                 id="input"
                 type="file"
-                class="mdc-text-field__input file"
-                accept=${this.accept}
+                class="file"
+                .accept=${this.accept}
+                .multiple=${this.multiple}
                 @change=${this._handleFilePicked}
                 aria-labelledby="label"
```

Comment:
Remove this - it's redundant with the `label` element and only points to the secondary text.